### DISCUSSION TITLE
Suggestion: provide a way to cancel a long running operation such as mz_zip_writer_add_path()

### DISCUSSION THREAD

**Discussion by ra-dave** [nmoinvaz/minizip#622] [Ideas]
In our app we use mz_zip_writer_add_path to zip up everything with one call.  The only problem with that is there is no way to cancel the operation and it can take several minutes when creating large zips.  I was hoping that returning an error code from the add_entry callback would in effect perform a cancel but alas the return value from the callback does nothing:

    if (writer->entry_cb != NULL)
        writer->entry_cb(handle, writer->entry_userdata, &writer->file_info);

We could traverse the folder hierarchy ourselves in order to handle the cancel but that is a lot of extra code.

**Comment by nmoinvaz** [comment]
Feel free to submit a PR.
